src/XrdPosix/XrdPosixMap.cc

```diff
@@ -80,6 +80,41 @@ mode_t XrdPosixMap::Flags2Mode(dev_t *rdv, uint32_t flags)
 
    return newflags;
 }
+
+/******************************************************************************/
+/*                             S t a t F i l l                                */
+/******************************************************************************/
+
+int XrdPosixMap::Entry2Buf(const XrdCl::DirectoryList::ListEntry &dirEnt, struct stat &buf)
+{
+   auto statInfo = dirEnt.GetStatInfo();
+   if (!statInfo) return EIO;
+
+   memset(&buf, '\0', sizeof(buf));
+   buf.st_mode   = XrdPosixMap::Flags2Mode(nullptr, statInfo->GetFlags());
```

Comment:
Why specifically state the class method you are calling in the same class where the call occurs? It makes it more difficult to read.

Reply:
Agreed.  This function was originally in `XrdPosixDir` but I refactored it to be in the same `XrdPosixMap` class.

Fixed.